File: TodoListApp/ClientApp/src/components/TodoItem.js
import React, { Component } from 'react';
import moment from 'moment';

import './style.css';

class TodoItem extends Component {

    tableBody = () => {
        const { todos, updateTodoItemFn } = this.props;
        console.log(todos);

        return todos.map((_todo, _index) => {
            return (
                <tr key={_todo.id}
                    className={_todo.status === "Close" ? "todoItem completed" : "todoItem"}
                    id={`todoItem-${_todo.id}`}
                    onClick={(e) => updateTodoItemFn(_todo.id)}
                >
                    <td>{_todo.id}</td>
                    <td>{_todo.description}</td>
                    <td>{moment(_todo.creationTime).format('llll')}</td>
                    <td>{_todo.status === "Close" ? moment(_todo.completionTime).format('llll') : "TBC"}</td>
                </tr>
            )
        })
    }

    render() {
        return (
            <div>
                <table className='table table-striped'>

                    <thead>
                        <tr>
                            <th scope="col">#</th>
                            <th scope="col">Todo Item</th>
                            <th scope="col">Start Time</th>
                            <th scope="col">Completion Time</th>
                        </tr>
                    </thead>

                    <tbody>
                        {this.tableBody()}
                    </tbody>
                </table>
            </div>
        );
    }
}

export default TodoItem;

File: TodoListApp/Models/TodoItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TodoListApp.Models
{
    public class TodoItem
    {
        // Primary Key for the DB
        public int Id { get; set; }
        // Todo item name
        public string Description { get; set; }
        // Status of the todo item [Open or Close]
        public string Status { get; set; }
        // Time when the todo item was created
        public DateTime CreationTime { get; set; }
        // Time when the todo item was completed
        public DateTime CompletionTime { get; set; }
    }
}
